### PR TITLE
Implement from_scratch option

### PR DESCRIPTION
**Pull request by tummychow**
Okay well, I asked for advice in #934, but I accidentally implemented it before getting an answer. Whoops!

Feel free to criticize style, implementation details, naming, tests/lack thereof, etc.

### CODE REVIEW

**Comment by tummychow** [comment]
Ping on this? @sampsyo ?

**Comment by jackwilsdon** [comment]
That looks good to me, although I'll let @sampsyo have the final call 😂 👍

File: docs/reference/config.rst [hunk 1]
```diff
@@ -475,6 +475,15 @@ Either ``yes`` or ``no``, controlling whether imported directories are
 recorded and whether these recorded directories are skipped.  This
 corresponds to the ``-i`` flag to ``beet import``.
 
+.. _from_scratch:
+
+from_scratch
+~~~~~~~~~~~~
+
+Either ``yes`` or ``no`` (default), controlling whether existing metadata is
+discarded when a match is applied. This corresponds to the ``-from_scratch``
```
Comment (sampsyo) [inline docs/reference/config.rst]: Tiny typo: missing one dash in the option name.

**Comment by sampsyo** [comment]
Sorry for the long delay in reviewing this! Thanks for waiting while it got to the top of my list.

This looks great: it's simple and it matches what users have been asking for. While we're here, I want to briefly take stock of all the ways we have for "clearing out metadata," which have grown steadily:

* The `scrub` plugin deletes all metadata and writes it back again, which removes **tags that beets doesn't know about** that show up in files' metadata.
* The `zero` plugin can selectively remove some metadata fields, or (in "whitelist" mode) only keep certain fields.
* This new feature is sort of like configuring `zero` to keep only the fields that beets fetches from remote servers.

Some users might want to configure *both* `scrub` and this new `from_scratch` option, or *also* enable `zero` to turn off some fields that come from remote sources.

It might be worth documenting this set of choices centrally somewhere, if anyone has a bright idea about where this should live.

**Comment by sampsyo** [comment]
Merged! :sparkles: Thank you again for the clean, simple implementation and the readable documentation.

**Comment by tummychow** [comment]
My pleasure, thank you for merging.

For myself, I actually use all three of scrub, zero, and from_scratch (zero for music that I import as-is, where from_scratch wouldn't apply). I personally think the detailed documentation on scrub/zero is sufficient for me to understand what they do and why I would want them, but my perspective might be skewed after thinking about and implementing this feature.

**Pull request merged**